corehq/tabs/tabclasses.py

```diff
@@ -59,9 +64,42 @@ def sidebar_items(self):
             request=self._request, domain=self.domain)
         custom_reports = CustomProjectReportDispatcher.navigation_sections(
             request=self._request, domain=self.domain)
-        sidebar_items = tools + report_builder_nav + custom_reports + project_reports
+        sidebar_items = tools + report_builder_nav + self._regroup_sidebar_items(custom_reports + project_reports)
         return self._filter_sidebar_items(sidebar_items)
 
+    def _regroup_sidebar_items(self, sidebar_items):
+        try:
+            ordering = ReportsSidebarOrdering.objects.get(domain=self.domain)
+        except ObjectDoesNotExist:
+            return sidebar_items
+
+        reports_to_move = {}  # report class name to SidebarPosition mapping
+        for heading, reports in ordering.config:
+            for i, report_class_name in enumerate(reports):
```

Comment:
I suspect this would just cause every report page in the domain to 500 if someone configures the ordering wrong right? seems like a bad failure mode. so i think should either add error handling here or add the validation going into the DB above